gradebookng/tool/src/java/org/sakaiproject/gradebookng/tool/pages/GradebookPage.java

```diff
@@ -390,7 +393,15 @@ public String getCssClass() {
 
 				};
 
-				cols.add(column);
+				if (settings.isCategoriesEnabled()) {
+					// insert category column after assignments in that category
+					currentColumnIndex = currentColumnIndex + category.getAssignmentList().size();
```

Comment:
Surprisingly yes! Because the assignments are already sorted by their category, all the uncategorised columns are at the end of the array.  This only iterates over the real categories and pokes the category average column into the right spot of the already sorted cols array.